Add Package Requirements option to Local Runner

john-jac opened the pull request:
*Issue #, if available:*

 Add Package Requirements option to Local Runner

*Description of changes:*

This command line option will use pip3 to download dependencies to the plugins folder and generate a local-requirements.txt file which will allow users to automatically create local WHL files to use with offline MWAA deployments such as private web servers and no Pypi access.



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

john-jac committed:
Add Package Requirements option to Local Runner

This command line option will use pip3 to download dependencies to the plugins folder and generate a local-requirements.txt file which will allow users to automatically create local WHL files to use with offline MWAA deployments such as private web servers and no Pypi access.

john-jac committed:
Updated to reflect last state of 2.2

john-jac committed:
Update mwaa-local-env

john-jac committed:
Fixed double file listing

Kalhan22 reviewed (approved):
We should also add information or steps to the read me to explain the intended nature of this change and how can customers use this to zip up the contents of plugins?

We should also specify the web server configuration(aka private) this is most suited for in the read me.

john-jac committed:
Improved method using --no-index and --find-links

Improved method using --no-index and --find-links without --no-deps makes it easier to ensure all dependencies are downloaded

john-jac commented:
> We should also add information or steps to the read me to explain the intended nature of this change and how can customers use this to zip up the contents of plugins?
> 
> We should also specify the web server configuration(aka private) this is most suited for in the read me.

Added to package-requirements the ability to auto-create the plugins.zip file

john-jac committed:
Added creation of plugins.zip
